internal/wasm/global.go

```diff
@@ -44,82 +36,16 @@ func (g *mutableGlobal) String() string {
 	}
 }
 
-type globalI32 uint64
-
-// compile-time check to ensure globalI32 is a api.Global
-var _ api.Global = globalI32(0)
-
-// Type implements the same method as documented on api.Global.
-func (g globalI32) Type() api.ValueType {
-	return ValueTypeI32
-}
-
-// Get implements the same method as documented on api.Global.
-func (g globalI32) Get() uint64 {
-	return uint64(g)
-}
-
-// String implements fmt.Stringer
-func (g globalI32) String() string {
-	return fmt.Sprintf("global(%d)", g)
-}
-
-type globalI64 uint64
-
-// compile-time check to ensure globalI64 is a api.Global
-var _ api.Global = globalI64(0)
-
-// Type implements the same method as documented on api.Global.
-func (g globalI64) Type() api.ValueType {
-	return ValueTypeI64
-}
-
-// Get implements the same method as documented on api.Global.
-func (g globalI64) Get() uint64 {
-	return uint64(g)
-}
-
-// String implements fmt.Stringer
-func (g globalI64) String() string {
-	return fmt.Sprintf("global(%d)", g)
-}
-
-type globalF32 uint64
-
-// compile-time check to ensure globalF32 is a api.Global
-var _ api.Global = globalF32(0)
-
-// Type implements the same method as documented on api.Global.
-func (g globalF32) Type() api.ValueType {
-	return ValueTypeF32
-}
-
-// Get implements the same method as documented on api.Global.
-func (g globalF32) Get() uint64 {
-	return uint64(g)
-}
-
-// String implements fmt.Stringer
-func (g globalF32) String() string {
-	return fmt.Sprintf("global(%f)", api.DecodeF32(g.Get()))
-}
-
-type globalF64 uint64
-
-// compile-time check to ensure globalF64 is a api.Global
-var _ api.Global = globalF64(0)
-
-// Type implements the same method as documented on api.Global.
-func (g globalF64) Type() api.ValueType {
-	return ValueTypeF64
+// mutableGlobal extends constantGlobal to allow updates.
+type mutableGlobal struct {
+	constantGlobal
```

Comment:
Expressing `mutableGlobal` in terms of `constantGlobal` is tempting to remove a bit of code but it's also a bit misleading.

The `String` function is main piece of logic that is really worth reusing, how about factoring this out in a function and keeping the implementations of `mutableGlobal` and `constantGlobal` separated?
```go
func (g *GlobalInstance) String() string {
  ...
}
```
```go
func (g constantGlobal) String() string {
    return g.g.String()
}
```
```go
func (g mutableGlobal) String() string {
    return g.g.String()
}
```

Reply:
Fair enough. It did feel a little weird replacing it everywhere.